In [14]:
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import LogisticRegression

In [13]:
# Import the data
data = pd.read_csv("Resources/lending_data.csv")
df = pd.DataFrame(data)
df.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0

In [4]:
# Check whether the data is clean
df.isnull().sum()

loan_size           0
interest_rate       0
borrower_income     0
debt_to_income      0
num_of_accounts     0
derogatory_marks    0
total_debt          0
loan_status         0
dtype: int64

## Split the data into training and testing sets


In [5]:
# Split the data into X_train, X_test, y_train, y_test
X = df.drop(columns="loan_status")
Y = df["loan_status"]
X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=0.2, random_state=42)

## LogisticRegression model

In [15]:
# Create a Logistic Regression Model
classifier = LogisticRegression()
classifier

LogisticRegression()

In [17]:
# Create and Train a Logistic Regression model print the model score
logistic_regression = classifier.fit(X_train, y_train)


print(f"Testing Data Score: {logistic_regression.score(X_test, y_test)}")
print(f"Training Data Score: {logistic_regression.score(X_train, y_train)}")

Testing Data Score: 0.9928424039205571
Training Data Score: 0.9915522022312504


In [11]:
# Train a Random Forest Classifier model and print the model score
from sklearn.ensemble import RandomForestClassifier
rf_model = RandomForestClassifier(n_estimators=128, random_state=78)
rf_model = rf_model.fit(X_train, y_train)

In [12]:
# print the model score
print(f"Testing Data Score: {rf_model.score(X_test, y_test)}")
print(f"Training Data Score: {rf_model.score(X_train, y_train)}")

Testing Data Score: 0.9924555068351818
Training Data Score: 0.9970658412329916


## Compare the two models

##### In testing the model, the Random Forest Classifier model has a lower score which is 0.9924 than the Logistic Regression model which is 0.9928. The Random Forest Classifier model is more accurate than the Logistic Regression model. In training the model, the Random Forest Classifier model has a higher score which is 0.9970 than the Logistic Regression model which is 0.9915.